cirq/protocols/json.py

```diff
@@ -204,14 +212,44 @@ def default(self, o):
         if isinstance(o, sympy.Symbol):
             return obj_to_dict_helper(o, ['name'], namespace='sympy')
 
+        if isinstance(o, (sympy.Add, sympy.Mul, sympy.Pow)):
+            return obj_to_dict_helper(o, ['args'], namespace='sympy')
+
+        if isinstance(o, sympy.Integer):
+            return {'cirq_type': 'sympy.Integer', 'i': o.p}
+
+        if isinstance(o, sympy.Float):
+            return {'cirq_type': 'sympy.Float', 'approx': float(o)}
+
+        if isinstance(o, sympy.Rational):
+            return {
+                'cirq_type': 'sympy.Rational',
+                'p': o.p,
+                'q': o.q,
+            }
+
+        if isinstance(o, pd.MultiIndex):
+            return {
+                'cirq_type': 'pandas.MultiIndex',
+                'tuples': list(o),
+                'names': list(o.names),
+            }
+
+        if isinstance(o, pd.Index):
```

Comment:
pandas has pretty robust support for json serialization of data-frames. You might want to investigate if you can just use that

Reply:
I looked into it. The `to_json` method on `DataFrame` does not include metadata such as the names of the columns in the index.

```
x = pd.DataFrame(index=pd.MultiIndex.from_tuples([(0, 0)], 
                                                      names=['x', 'y']), 
                      data=[[0, 0]], 
                      columns=pd.Index(['a', 'b'], name='c'))
```

```
print(x)
```

```
c    a  b
x y      
0 0  0  0
```

```
print(x.to_json())
```

```
'{"a":{"[0,0]":0},"b":{"[0,0]":0}}'
```

Reply:
oh wild